### ISSUE TITLE
Kick a user from lobby?

### ISSUE THREAD

**Issue by nick-morhun** [Facepunch/Facepunch.Steamworks#92]
Is this feature hidden somewhere or will it be implemented in future?

**Comment by mickwonnink** [comment]
@nick-morhun Sorry, not an answer to your question, but GIT has no way of contacting/messaging someone. 

I saw your post yesterday about how you got Lobby.Create to work and was wondering if you had to do anything specific to accomplish this? As this does not work for me. I have the Client working but Lobby.Create doesn't seem to create a valid lobby. If you can help please message me on: mick.wonnink@live.nl

**Comment by nick-morhun** [comment]
How to trigger the event Lobby.OnLobbyStateChanged with Lobby.MemberStateChange.Kicked parameter?

**Comment by garrynewman** [comment]
I had a quick scan, I can't see any way to kick people from lobbies in the Steam SDK. Anyone know any differently?

**Comment by nick-morhun** [comment]
Hi, since then I have found the answer on Steam forums: you just send a chat message to kick, and the user whose ID you called out leaves.

**Comment by LoneSurivor** [comment]
@nick-morhun Do you have the link or an example of the message to kick the user from the lobby?

**Comment by WaGi-Coding** [comment]
anybody? ^^

**Comment by WaGi-Coding** [comment]
> Hi, since then I have found the answer on Steam forums: you just send a chat message to kick, and the user whose ID you called out leaves.

the whole point of kicking somebody is prob. because they failed to auth on the actual socketconnection. And if you rely on a leave from the member who should get kicked, he can simply avoid that by modding the game or writing his own script

**Comment by SametHope** [comment]
Bump

**Comment by WaGi-Coding** [comment]
> Bump

why the bump? This is a Valve issue after all. Thats why games using steam lobbies are also insecure and buggy as hell ^^
Valve does not offer a way to kick a member from the steam lobby, so you cannot really kick an unauthorized user from the steam lobby itself. Means you also cannot really rely on the lobby max members for your lobby member count status.

You are forced to make a dirty workaround, but you just cannot kick someone from a steam lobby, so if you kick them from the server because they fail to auth properly, you can kick them from your server connection, but they will remain in you steam lobby.

In short: Valve does not care about simple concepts and let their API dust in the wild quality- and security-wise, which is insane when you think about how rich this company is. But at least they only charge you once 100$ per appId for using their 2000's  like services
